### PR TITLE
Simplify mapping

### PR DESCRIPTION
**Pull request by fsouza**
Simplifying here means removing a lot of the stuff that we
added to be able to handle some of NYT's business logic, we'll move it
into a closed source piece. To support that, the actual mapping now
happens in a library that can be used independently from the service
itself. The service still works just as fine, but it's waaay simpler.

Mapping in gcs-helper only needs two things: a prefix and a regular
expression to filter the objects that match the filter.

Extra resources, additional prefixes and multiple regexp filters should
not be part of the core of gcs-helper.

### CODE REVIEW

File: vodmodule/mapper.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import (
+	"context"
+	"path"
+	"regexp"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+)
+
+const maxTry = 5
```
Comment (scentless-apprentice) [inline vodmodule/mapper.go]: Not a big deal, but could we rename this to `maxTries`?

File: map_test.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"net/http"
+	"testing"
+	"time"
+)
+
+func TestServerMapBucketNotFound(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (fsouza) [reply map_test.go]: The one in ``server_test.go`` covers it, but I'll add another one here with a clearer name.

**Comment by fsouza** [comment]
@scentless-apprentice please take another look.

File: vodmodule/mapper.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import (
+	"context"
+	"path"
+	"regexp"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+)
+
+const maxTries = 5
```
Comment (scentless-apprentice) [inline vodmodule/mapper.go]: 🙏

**Pull request merged**